### PR TITLE
Updating FastAPI Docs with a python-ngrok example

### PR DESCRIPTION
**Pull request by firecube-oss**
* This PR adds an example of how to use Ngrok with the ngrok-python SDK

### CODE REVIEW

File: docs/using-ngrok-with/fastAPI.md [hunk 1]
```diff
@@ -1,7 +1,66 @@
...
+# ngrok free tier only allows one agent. So we tear down the tunnel on application termination
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    logger.info("Setting up Ngrok Tunnel")
+    ngrok.set_auth_token(NGROK_AUTH_TOKEN)
+    ngrok.forward(
+        addr=APPLICATION_PORT,
+        labels=NGROK_EDGE,
+        proto="labeled",
+   )
```
Comment (nijikokun) [inline docs/using-ngrok-with/fastAPI.md:48]:
```suggestion
    )
```

File: docs/using-ngrok-with/fastAPI.md [hunk 2]
```diff
@@ -1,7 +1,66 @@
 ---
-title: Fast API
+title: FastAPI
 ---
 
-# Using ngrok with FastAPI
+# Option 1: Using ngrok-python
```
Comment (salilsub) [inline docs/using-ngrok-with/fastAPI.md:5]:
```suggestion
# Using ngrok with FastAPI
```

File: docs/using-ngrok-with/fastAPI.md [hunk 3]
```diff
@@ -1,7 +1,66 @@
...
+async def root():
+    return {"message": "Hello World"}
+
+if __name__ == "__main__":
+    uvicorn.run("main:app", host="127.0.0.1", port=APPLICATION_PORT, reload=True)
+```
+
+# Option 2: Using pyngrok
+
+If you're looking to natively embed the ngrok agent into your FastAPI application, you can leverage the [pyngrok project](https://pyngrok.readthedocs.io/en/latest/integrations.html#fastapi) to start a tunnel anytime you start the FastAPI Server via uvicorn.
```
Comment (salilsub) [inline docs/using-ngrok-with/fastAPI.md:66]:
```suggestion
```

File: docs/using-ngrok-with/fastAPI.md [hunk 4]
```diff
@@ -1,7 +1,66 @@
 ---
-title: Fast API
+title: FastAPI
 ---
 
-# Using ngrok with FastAPI
+# Option 1: Using ngrok-python
 
-If you're looking to natively embed the ngrok agent into your Fast API application, you can leverage the [pyngrok project](https://pyngrok.readthedocs.io/en/latest/integrations.html#fastapi) to start a tunnel anytime you start the FastAPI Server via uvicorn.
+You can leverage the [ngork-python](https://github.com/ngrok/ngrok-python) library to embed the ngrok agent in to FastAPI Applications.
```
Comment (salilsub) [inline docs/using-ngrok-with/fastAPI.md:7]:
```suggestion
You can leverage the [ngrok-python](https://github.com/ngrok/ngrok-python) library to embed the ngrok agent in to FastAPI Applications. 
```

File: docs/using-ngrok-with/fastAPI.md [hunk 5]
```diff
@@ -1,7 +1,66 @@
...
 ---
 
-# Using ngrok with FastAPI
+# Option 1: Using ngrok-python
 
-If you're looking to natively embed the ngrok agent into your Fast API application, you can leverage the [pyngrok project](https://pyngrok.readthedocs.io/en/latest/integrations.html#fastapi) to start a tunnel anytime you start the FastAPI Server via uvicorn.
+You can leverage the [ngork-python](https://github.com/ngrok/ngrok-python) library to embed the ngrok agent in to FastAPI Applications. 
+
+:::warning
+  ngork-python, Pydantic and FastAPI libraries used in this example are sensitive to versions.  Use of Virtual Environments is highly recommended
```
Comment (salilsub) [inline docs/using-ngrok-with/fastAPI.md:10]:
```suggestion
  ngrok-python, Pydantic and FastAPI libraries used in this example are sensitive to versions.  Use of Virtual Environments is highly recommended 
```

**Review by salilsub — CHANGES_REQUESTED** [review]
Using `pyngrok` shouldn't be an option 2 since it isn't supported by ngrok. Using the SDK is the way to go, it should be the only option recommended in docs. If the suggested changes are made, will approve and merge.

**Review by russorat — APPROVED** [review]
lgtm! thank you!

**Pull request merged**